### PR TITLE
Unsafe module2

### PR DESCRIPTION
**Pull request by jule64**
Same as before but with improved commit messages

(this is the second try for this branch as first time build failed due to code in doc failing)

### CODE REVIEW

File: unsafe/shared/src/main/scala/monocle/unsafe/UnsafeSelect.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package monocle.unsafe
+
+import monocle.Prism
+
+
+/**
+  * Created by Cesar on 16/07/2016.
```
Comment (julien-truffaut) [inline unsafe/shared/src/main/scala/monocle/unsafe/UnsafeSelect.scala]: would you mind to remove this comment

File: test/shared/src/test/scala/monocle/unsafe/UnsafeSelectSpec.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import monocle.MonocleSuite
+import monocle.law.discipline.{OptionalTests, PrismTests}
+import monocle.macros.GenLens
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+
+import scalaz.Equal
+
+/**
+  * Created by Cesar on 16/07/2016.
```
Comment (julien-truffaut) [inline test/shared/src/test/scala/monocle/unsafe/UnsafeSelectSpec.scala]: same here

File: core/shared/src/main/scala/monocle/Traversal.scala [hunk 3]
```diff
@@ -248,6 +248,22 @@ object Traversal {
...
+
+  /**
+    * Composes N lenses horizontally.  Note that although it is possible to pass two or more lenses
+    * that point to the same `A`, in practice it is considered an unsafe usage of the method (see https://github.com/julien-truffaut/Monocle/issues/379#issuecomment-236374838).
+    */
+  def applyN[S, A](xs: Lens[S, A]*): Traversal[S, A] = {
+    val lenses = xs.toList
+    new PTraversal[S, S, A, A] {
+      def modifyF[F[_] : Applicative](f: A => F[A])(s: S): F[S] = {
+        val as: List[(F[A], Lens[S, A])] = lenses.map(lens => (f(lens.get(s)), lens))
```
Comment (julien-truffaut) [inline core/shared/src/main/scala/monocle/Traversal.scala]: I believe we can slightly simply and improve the performance of this function using the following impl:
```
xs.foldLeft(Applicative[F].pure(s))((fs, lens) => 
  Applicative[F].apply2(f(lens.get(s)), fs)((a, s) => lens.set(a, s))
)
```

File: docs/src/main/tut/unsafe_module.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+title:  "Unsafe Module"
+section: "optics"
+pageSource: "https://raw.githubusercontent.com/julien-truffaut/Monocle/master/docs/src/main/tut/unsafe_module.md"
+---
+
+# Intro
+
+The `unsafe` module contains `Optics` that cannot fully satisfy the set of `Optics` `Laws` of the `core` module.
+
+While these "unsafe" `Optics` are still handy for many common use case, care must be taken by the user to avoid the "unlawful" usage as it is not enforced by the library.
```
Comment (julien-truffaut) [inline docs/src/main/tut/unsafe_module.md]: nice doc 👍

**Comment by julien-truffaut** [comment]
It looks great thanks @jule64 and @cesartl, I just made a few minor comments.

File: core/shared/src/main/scala/monocle/Traversal.scala (hunk 3 above)
Comment (jule64) [inline core/shared/src/main/scala/monocle/Traversal.scala]: Thanks for the review, I'll add the fixes tomorrow

Comment (jule64) [inline core/shared/src/main/scala/monocle/Traversal.scala]: I am going to push the update now.  Minor detail here but I had to add `(s)` to `lens.set(a,s)` for the improved impl to run

Comment (jule64) [inline core/shared/src/main/scala/monocle/Traversal.scala]: I mean lens.set(a)(s)

Comment (julien-truffaut) [inline core/shared/src/main/scala/monocle/Traversal.scala]: correct

**Comment by julien-truffaut** [comment]
👍

**Pull request merged**